### DISCUSSION TITLE
[V5] SkipLocked for sql server

### DISCUSSION THREAD

**Discussion by viniciusnazario** [adonisjs/core#2671] [Help]
Hello guys, Is there skipLocked for the sql server?  Or something like that in query builder? 
I was unable to add "no lock" to the query builder. Would someone have an example please.

**Comment by RomainLanz** [comment]
Hey @viniciusnazario! 👋 

The `skipLocked` method only work for MySQL 8.0+ and PostgreSQL 9.5+.
It seems that [there is a way to add something like Advisory Lock to MSSQL](https://dba.stackexchange.com/questions/176424/does-microsoft-sql-server-offer-an-advisory-locks-feature-like-postgres), I would recommend you to create an issue inside the [Knex repository](https://github.com/knex/knex/issues).

**Reply by viniciusnazario** [reply]
Hey @RomainLanz

Thanks for the answer, I will open a issue within in the knex repository as recommended, thank you!

**Comment by thetutlage** [comment]
I think there is already a skipLocked method https://docs.adonisjs.com/reference/database/query-builder#skiplocked

**Reply by RomainLanz** [reply]
As per Knex doc, this method only work for MySQL and PostgreSQL, not for MSSQL.

**Reply by thetutlage** [reply]
Ahhh, so we have mentioned mssql too in our docs?

**Reply by RomainLanz** [reply]
No we have not, our doc also state that it is working only with MySQL and PostegreSQL.
The thing is, it seems that there is a way to make it work with MSSQL, but I believe Knex should work on that.

@viniciusnazario, have you created an issue on Knex repository?

**Reply by viniciusnazario** [reply]
Hello there. I apologize for the delay. I opened today, and I am waiting to answer, as soon as I have something new posted here. Thank you for your attention!
